index.js

```diff
@@ -0,0 +1,104 @@
+const lazyloadOptions = {
+  forceIntersectionObserver: false,
+  forceNativeLazyload: false,
+  nativeLazyloadEnabled: true,
+
+  getObserverOptions: () => ({
+    root: null,
+    rootMargin: `0px 0px ${window.innerHeight / 2}px 0px`,
+    threshold: 1
+  })
+};
+
+let observer = null;
+const getElements = (selector = '[data-lazy]') => Array.from(document.querySelectorAll(selector));
+
+const setSource = element => {
+  const { src, srcset } = element.dataset;
+
+  if (src) {
+    element.src = src;
+  }
+
+  if (srcset) {
+    element.srcset = srcset;
+  }
+
+  element.removeAttribute('data-lazy');
+};
+
+const onIntersect = (entries, entryObserver) => {
+  entries.forEach(entry => {
+    if (entry.isIntersecting) {
+      const { src, srcset, wait } = entry.dataset;
+
+      if (wait && entry.tagName === 'IMG') {
+        const image = new Image();
+        entry.onload = () => setSource(entry);
+
+        if (src) {
+          image.src = src;
+        }
+
+        if (srcset) {
+          image.srcset = srcset;
+        }
+      } else {
+        if (wait && entry.tagName === 'IFRAME') {
+          entry.onload = () => setSource(entry);
+        }
+
+        setSource(entry);
+      }
+
+      entryObserver.unobserve(entry);
+    }
+  });
+};
+
+const init = () => {
```

Comment:
I'd like for init to return the `observableElements` if it can't handle it. If we were to drop this on LeanIN we would like to set scroll-triggers for fallback or maybe just provide any other alternative.